cadl-extension/src/code-model-builder.ts

```diff
@@ -181,9 +185,23 @@ export class CodeModelBuilder {
   }
 
   public build(): CodeModel {
-    ignoreDiagnostics(getAllHttpServices(this.program)).map((service) =>
-      service.operations.map((it) => this.processRoute(it)),
-    );
+    const clients = listClients(this.program);
+    // TODO: multiple clients
+    for (const client of clients) {
+      const operationGroups = listOperationGroups(this.program, client);
+
+      const operationWithoutGroup = listOperationsInOperationGroup(this.program, client);
+      for (const operation of operationWithoutGroup) {
+        this.processRoute("", operation);
+      }
+
+      for (const operationGroup of operationGroups) {
+        const operations = listOperationsInOperationGroup(this.program, operationGroup);
+        for (const operation of operations) {
+          this.processRoute(operationGroup.type.name, operation);
+        }
+      }
+    }
```

Comment:
major change here.

for now, it still uses m4 format. will update it to a new format that supports multiple clients.